base_layer/wallet/src/output_manager_service/mod.rs

```diff
@@ -20,5 +20,77 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+use crate::output_manager_service::{handle::OutputManagerHandle, service::OutputManagerService};
+
+use futures::{future, Future};
+use log::*;
+use tari_core::types::PrivateKey;
+use tari_service_framework::{
+    handles::ServiceHandlesFuture,
+    reply_channel,
+    ServiceInitializationError,
+    ServiceInitializer,
+};
+use tokio::runtime::TaskExecutor;
 pub mod error;
-pub mod output_manager_service;
+pub mod handle;
+pub mod service;
+
+const LOG_TARGET: &'static str = "wallet::output_manager_service::initializer";
+
+pub struct OutputManagerServiceInitializer {
+    master_key: Option<PrivateKey>,
+    branch_seed: Option<String>,
+    primary_key_index: Option<usize>,
+}
+
+impl OutputManagerServiceInitializer {
+    pub fn new(master_key: PrivateKey, branch_seed: String, primary_key_index: usize) -> Self {
+        Self {
+            master_key: Some(master_key),
+            branch_seed: Some(branch_seed),
+            primary_key_index: Some(primary_key_index),
+        }
+    }
+}
+
+impl ServiceInitializer for OutputManagerServiceInitializer {
+    type Future = impl Future<Output = Result<(), ServiceInitializationError>>;
+
+    fn initialize(&mut self, executor: TaskExecutor, handles_fut: ServiceHandlesFuture) -> Self::Future {
+        let master_key = self
+            .master_key
+            .take()
+            .expect("Output Manager Service initializer already called");
+        let branch_seed = self
+            .branch_seed
+            .take()
+            .expect("Output Manager Service initializer already called");
+        let primary_key_index = self
+            .primary_key_index
+            .take()
+            .expect("Output Manager Service initializer already called");
+
+        let (sender, receiver) = reply_channel::unbounded();
+
+        let oms_handle = OutputManagerHandle::new(sender);
+
+        // Register handle before waiting for handles to be ready
+        handles_fut.register(oms_handle);
+        executor.spawn(async move {
+            let _handles = handles_fut.await;
```

Comment:
If your service doesn't require any handles you don't have to await on it - your service can start straight away

Reply:
Oh right! Done.